generic/tests/linux_stress.py

```diff
@@ -0,0 +1,208 @@
+
+from virttest import data_dir
+from virttest import utils_misc
+from virttest import error_context
+
+
+class Stress(object):
+    """
+    Stress utility for linux guest
+    """
+
+    def __init__(self, params, vm):
+        """
+        Initialize StressTest object
+
+        :param params: Dictionary with the test parameters
+        :param vm: Object qemu_vm.VM, the guest need stress
+        """
+        self.params = params
+        self.dst_path = "/home"
+        self.tarball_name = 'stress-1.0.4'
+        self.downloaded = False
+        self.vm = vm
+
+    @error_context.context_aware
+    def download_and_uncompress_stress(self):
+        """
+        Download and uncompress the stress tarball
+        """
+        download_url = self.params.get('download_url',
+                                       'http://people.seas.harvard.edu/'
+                                       '~apw/stress/stress-1.0.4.tar.gz')
+
+        tarball_path = '%s/%s.tar.gz' % (data_dir.get_data_dir(),
+                                         self.tarball_name)
+        pkg_md5 = self.params.get('pkg_md5', '890a4236dd1656792f3ef9a190cf99ef')
+        error_context.context('Download stress tarball', logging.info)
+        tarball_path = download.get_file(
+            download_url, tarball_path, hash_expected=pkg_md5)
+
+        error_context.context('Copy stress tarball to guest', logging.info)
+        self.vm.copy_files_to(tarball_path, self.dst_path)
+        tarball_path = '%s/%s.tar.gz' % (self.dst_path, self.tarball_name)
+
+        uncompress_cmd = self.params.get('uncompress_cmd', 'tar xvfz %s -C %s')
+        uncompress_cmd = uncompress_cmd % (tarball_path, self.dst_path)
+        error_context.context('Uncompress the stress tarball', logging.info)
+        session = self.vm.wait_for_login()
+        if session.cmd_status(uncompress_cmd) == 0:
+            session.cmd_output_safe("rm -f %s" % tarball_path)
+        session.close()
+        self.dst_path = '%s/%s' % (self.dst_path, self.tarball_name)
+        self.downloaded = True
+
+    @error_context.context_aware
+    def install(self):
+        """
+        To download, abstract, build and install the stress
+
+        :return: True if installation success, otherwise False
+        """
+        if self.check_installed():
+            error_context.context('stress has already been installed.')
+            return True
+        else:
+            error_context.context('stress has not been installed, '
+                                  'download and install...')
+            self.download_and_uncompress_stress()
+            session = self.vm.wait_for_login()
+            session.cmd_output_safe('cd %s' % self.dst_path)
+            make_cmd = self.params.get(
+                'make_cmd', './configure && make && make install')
+            error_context.context('make and install the stress app', logging.info)
+            session.cmd_output_safe(make_cmd)
+            session.close()
```

Comment:
Why not use a `self.session()`?

Reply:
@sitoliu Cause this class is designed as its functions can be called separately, not sure where the user will stop the session(means hard to decide when to call self.session.close()). So better open up a separate session in each function and close it immediately just before leave the function itself.

Reply:
Thanks, got it.